### DISCUSSION TITLE
Using TCP in specific situations

### DISCUSSION THREAD

**Discussion by Artnal** [pivpn/pivpn#1437] [Q&A] [answered]
Hi there, 

Apologies if the question is stupid, I'm fairly quickly out of my depth on these topics. 

I've been running Wireguard thanks to pivpn of my Pi for a couple of years, which allows me to use my pihole setup from home. 

While travelling recently, I've been faced with a situation where my hotel wouldn't allow any connection through Wireguard. I ultimately understood that it had probably blocked UDP traffic, and I'm looking for a way to prevent this situation in the future. I understood that there would be two main ways: 
1) Setting up a TCP workaround for Wireguard; 
2) Using OpenVPN 

For the first option, it seems that the current sitaution is [not satisfactory for Android users](https://www.reddit.com/r/WireGuard/comments/p6123n/wireguard_over_tcp_reviewing_existing_options_to/). I'm not sure how I would go around implementing it. 

For the second option, since Wireguard is perfectly functional 99% of the time, I'd simply need OpenVPN as a backup. Is there a way to have both OpenVPN and Wireguard installed through pivpn on the same Pi?

Thanks a bunch, I hope the question makes sense. 

Cheers

**Comment by orazioedoardo** [comment, marked as answer]
> Is there a way to have both OpenVPN and Wireguard installed through pivpn on the same Pi?

Yes, just run the script again and install the other VPN.

**Reply by Artnal** [reply]
I'm an idiot, thanks a bunch for your answer!
